app/assets/javascripts/controllers/container_topology/container_topology_controller.js

```diff
@@ -0,0 +1,30 @@
+angular.module('topologyApp', ['kubernetesUI'])
+.config(['$httpProvider', function($httpProvider) {
+    $httpProvider.defaults.headers.common['X-CSRF-Token'] = jQuery('meta[name=csrf-token]').attr('content');
+}])
+.controller('containerTopologyController', ['$scope', '$http', '$interval', "$location",  function($scope, $http, $interval, $location) {
+    $scope.refresh = function() {
+        var id;
+        if ($location.absUrl().match("show/$") || $location.absUrl().match("show$")) {
+           id = '';
+        }
+        else {
+            id = '/'+ (/container_topology\/show\/(\d+)/.exec($location.absUrl())[1]);
+        }
+
+        var url = '/container_topology/data'+id;
+        $http.get(url).success(function(data) {
+            $scope.items = data.data.items;
+            $scope.relations = data.data.relations;
+            $scope.kinds = data.data.kinds;
+        });
+
+    };
+
+    $scope.refresh();
+    $interval( $scope.refresh, 1000*60*3);
+
+    $scope.$on('$destroy', function() {
+        $interval.cancel;
```

Comment:
Almost right .. except `$interval.cancel` is a function - you need to pass it the return value of the original $interval call.. So `var promise = $interval( $scope.refresh, 1000*60*3);` on line 25 and `$interval.cancel(promise)` here.